iceoryx2-bb/log/src/lib.rs

```diff
@@ -271,9 +266,80 @@ pub fn get_logger() -> &'static dyn Log {
     }
 }
 
+#[cfg(feature = "std")]
+mod from_env {
+    use super::{set_log_level, LogLevel, DEFAULT_LOG_LEVEL};
+    use std::env;
+
+    fn get_log_level_from_str_fuzzy(
+        log_level_string: &str,
+        log_level_fallback: LogLevel,
+    ) -> LogLevel {
+        match log_level_string.to_lowercase().as_str() {
+            "trace" => LogLevel::Trace,
+            "debug" => LogLevel::Debug,
+            "info" => LogLevel::Info,
+            "warn" => LogLevel::Warn,
+            "error" => LogLevel::Error,
+            "fatal" => LogLevel::Fatal,
+            _ => {
+                eprintln!(
+                    "Invalid value for 'IOX2_LOG_LEVEL' environment variable!\
+                    \nFound: {log_level_string:?}\
+                    \nAllowed is one of: fatal, error, warn, info, debug, trace\
+                    \nSetting log level as : {log_level_fallback:?}"
+                );
+                log_level_fallback
+            }
+        }
+    }
+
+    /// Sets the log level by reading environment variable "IOX2_LOG_LEVEL" or default it with LogLevel::INFO
+    pub fn set_log_level_from_env_or_default() {
+        set_log_level_from_env_or(DEFAULT_LOG_LEVEL);
+    }
+
+    /// Sets the log level by reading environment variable "IOX2_LOG_LEVEL", and if the environment variable
+    /// doesn't exit it sets it with a user-defined logging level
+    pub fn set_log_level_from_env_or(v: LogLevel) {
+        let log_level = env::var("IOX2_LOG_LEVEL")
+            .ok()
+            .map(|s| get_log_level_from_str_fuzzy(&s, v))
+            .unwrap_or(v);
+        set_log_level(log_level);
+    }
+}
+
 #[doc(hidden)]
 pub fn __internal_print_log_msg(log_level: LogLevel, origin: Arguments, args: Arguments) {
     if get_log_level() <= log_level as u8 {
         get_logger().log(log_level, origin, args)
     }
 }
+
+// TODO(#1127): Add proper printing abstraction that can be implemented for no_std platforms
+#[cfg(feature = "std")]
+#[macro_export]
+macro_rules! println {
```

Comment:
Please rename it to `cout`

Reply:
While I like the idea on one hand, it might look out of place in a Rust codebase, which uses `*print*` all over the place. How about `iprintln` :grin: